### PR TITLE
feat: add possibility to set params for the client

### PR DESCRIPTION
**Pull request by oneslash**
This would be needed for us to put a proxy between Algolia and our frontend to track impressions more effectively on the backend.

**Summary**
I want to set up `hosts.read` and `hosts.write` to setup my proxy host, which will track product ids returned for a set of filters and etc.

Additionally, we will use more than one Algolia cluster so my `proxy` would balance between them without client modifications.

**Result**

### CODE REVIEW

**Comment by francoischalifour** [comment]
Thanks for the PR. As explained in the previous PR (https://github.com/algolia/angular-instantsearch/pull/711#pullrequestreview-364127484), this PR should target `master` if you want this change to land in Angular InstantSearch v2.

Adding these two tests in `create-ssr-algoia-client.spec.ts` would make sure that it behaves as expected:

```ts
it('forwards the default options to the search client', () => {
  const addAlgoliaAgent = jest.fn();
  algoliasearchProxy.mockImplementation(() => {
    return {
      addAlgoliaAgent,
    };
  });

  createSSRSearchClient({
    appId: 'appId',
    apiKey: 'apiKey',
    httpClient: null,
    HttpHeaders: null,
    makeStateKey: null,
    transferState: null,
  });

  expect(algoliasearchProxy).toHaveBeenCalledWith('appId', 'apiKey', {});
});

it('forwards the options to the search client', () => {
  const addAlgoliaAgent = jest.fn();
  algoliasearchProxy.mockImplementation(() => {
    return {
      addAlgoliaAgent,
    };
  });

  createSSRSearchClient({
    appId: 'appId',
    apiKey: 'apiKey',
    options: {
      queryParameters: {},
    },
    httpClient: null,
    HttpHeaders: null,
    makeStateKey: null,
    transferState: null,
  });

  expect(algoliasearchProxy).toHaveBeenCalledWith('appId', 'apiKey', {
    queryParameters: {},
  });
});
```

Would you be able to update this PR accordingly?

**Comment by oneslash** [comment]
@francoischalifour thank you very much, I am sorry, I am not a typescript professional nor javascript, thats why copied your tests :)

File: src/__tests__/create-ssr-algolia-client.spec.ts [hunk 1]
```diff
@@ -1,36 +1,51 @@
 import { createSSRSearchClient } from '../create-ssr-algolia-client';
 import * as algoliasearchProxy from 'algoliasearch/index';
-import { VERSION } from '../version';
-import { VERSION as AngularVersion } from '@angular/core';
 
 jest.mock('algoliasearch/index');
 
 describe('Create SSR', () => {
-  it('passes the User-Agent', () => {
```
Comment (francoischalifour) [inline src/__tests__/create-ssr-algolia-client.spec.ts]: We should keep this test since it tests another behavior.

Comment (oneslash) [reply src/__tests__/create-ssr-algolia-client.spec.ts]: oops, thank you :)

**Comment by francoischalifour** [comment]
@oneslash Would you be interested in porting this feature in the next major version too? (`v3` branch)

**Comment by oneslash** [comment]
@francoischalifour thank you very much. I can port to `v3`, should I target `v3` branch?

Regarding release, when do you think the release for `v2` would be?

**Comment by francoischalifour** [comment]
Yes, the changes for Angular InstantSearch 3 are on the [`v3` branch](https://github.com/algolia/angular-instantsearch/blob/v3/src/create-ssr-search-client.ts).

We can release a new version any time this week, depending on how urgent it is for you to have this feature in your code base.

**Comment by oneslash** [comment]
@francoischalifour this week should be great. thank you

**Comment by oneslash** [comment]
@francoischalifour an attempt for v3 here #715

**Comment by francoischalifour** [comment]
Released in [2.3.0](https://github.com/algolia/angular-instantsearch/blob/master/CHANGELOG.md#230-2020-04-03).

**Comment by oneslash** [comment]
merci @francoischalifour

**Pull request merged**